torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1786,6 +1786,13 @@ def _make_data_dependent_error(self, expr):
             # problem
         )
 
+    def _set_replacement(self, a: "sympy.Symbol", expr: "sympy.Expr"):
```

Comment:
I do not think setting replacements is the same as specialization?

Reply:
Right, this is just collecting all the cases that cause a replacement to be added / updated. Specialization is when they actually simplify to ints.

Reply:
But the flag name is `print_specializations` and the log comment is too :P

Reply:
To be clear, I'm only logging when `expr` is a int / float, see check right below.